lib/session_base.php

```diff
@@ -304,6 +304,22 @@ private function cookie_domain($request) {
         return $domain;
     }
 
+    /**
+     * @param Hm_Request $request request object
+     * @return string
+     */
+    private function cookie_path($request) {
+        $path = $this->site_config->get('cookie_path', false);
+        if (!$path) {
+            $path = $request->path;
+        }
+        if ($path == 'none') {
```

Comment:
Is this a good idea to check for a special value `none` (which is valid just for the server-side configuration) if it might be forged by the client (see line `314`)?

Reply:
I just followed the same way as cookie_domain variable is set. Indeed, if we end up with a URI path of 'none' we might default to an empty cookie path which will be a difference with the old code - it would have used 'none' as the cookie path. I don't mind removing this 'none' functionality from here. Just wanted to keep it consistent with cookie domain one.

Reply:
@jasonmunro what do you think?

Reply:
Sorry for the delay on this. @dumblob makes a good point, though I don't think there is any directly actionable attack vector as the result will likely be you get logged out if you forge 'none' for both cookie path or cookie domain. Regardless, I'm going to go ahead and merge this then fix both methods to eliminate that possibility.